src/System.Private.CoreLib/src/System/Object.CoreCLR.cs

```diff
@@ -16,7 +26,33 @@ public partial class Object
         // object.  This is always a shallow copy of the instance. The method is protected
         // so that other object may only call this method on themselves.  It is intended to
         // support the ICloneable interface.
-        [MethodImpl(MethodImplOptions.InternalCall)]
-        protected extern object MemberwiseClone();
+        protected unsafe object MemberwiseClone()
+        {
+            object clone = this;
+
+            AllocateClone(JitHelpers.GetObjectHandleOnStack(ref clone));
+
+            // copy contents of "this" to the clone
+
+            MethodTable* pMT = RuntimeHelpers.GetMethodTable(this);
+
+            // See comment on RawArrayData for details
+            nuint byteCount = pMT->BaseSize - (nuint)(2 * sizeof(IntPtr));
+            if (pMT->HasComponentSize)
```

Comment:
I wonder if this pattern will appear elsewhere in our codebase and if it would make sense to extract to a helper. Probably wait until we actually have a need for a shared helper, I guess?

Reply:
I have moved it to the helper. It looks better if nothing else.